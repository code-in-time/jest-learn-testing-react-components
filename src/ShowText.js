import React, { Component } from 'react';

class ShowText extends Component {

  state = {
    showText: true
  }

  toggleText = () => {
    this.setState( {showText: !this.state.showText} )
  }

  render() { 
    return ( 
      <div>
        testbcvbcvbc <button type="button" onClick={() => this.toggleText()}>click</button>

        <hr />

        { this.state.showText && <span className="text">this is text to show</span>}
      </div>
     );
  }
}
 
export default ShowText;