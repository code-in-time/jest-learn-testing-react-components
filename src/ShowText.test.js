import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ShowText from './ShowText';

Enzyme.configure({ adapter: new Adapter() });


describe('Test the show text component', () => {
  it('should test if there is text', () => {
    const wrapper = shallow(<ShowText/>);
    const texts = wrapper.find('span.text');
    expect(texts.text()).toBe('this is text to show')
  });

  it('should test if there is no text', () => {
    const wrapper = shallow(<ShowText/>);
    const button = wrapper.find('button');

    button.simulate('click');

    const texts = wrapper.find('span');
    expect(texts.length).toBe(0);
  });
});

