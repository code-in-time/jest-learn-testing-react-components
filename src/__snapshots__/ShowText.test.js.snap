// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matched snapshot 1`] = `
<div>
  testbcvbcvbc 
  <button
    onClick={[Function]}
    type="button"
  >
    click
  </button>
  <hr />
  <span
    className="text"
  >
    this is text to show
  </span>
</div>
`;
